modules/contracts/src.ts/services/ethService.ts

```diff
@@ -195,15 +196,22 @@ export class EthereumChainService extends EthereumChainReader implements IVector
     txFn: (gasPrice: BigNumber, nonce: number) => Promise<TransactionResponse | undefined>,
     gasPrice: BigNumber,
     signer: Signer,
+    chainId: number,
     nonce?: number,
   ): Promise<Result<TransactionResponse | undefined, Error>> {
     // Queue up the execution of the transaction.
     return await this.queue.add(
       async (): Promise<Result<TransactionResponse | undefined, Error>> => {
         try {
           // Send transaction using the passed in callback.
-          const actualNonce: number = nonce ?? (await signer.getTransactionCount("pending"));
+          const actualNonce: number =
+            nonce ?? this.nonces.get(chainId) ?? (await signer.getTransactionCount("pending"));
           const response: TransactionResponse | undefined = await txFn(gasPrice, actualNonce);
+          // After calling tx fn, set nonce to usedNonce + 1 IFF the
+          // nonce wasnt pre-specified (i.e. not re-sending tx)
+          if (typeof nonce === undefined) {
```

Comment:
why `typeof` ? Can't we just say `if (nonce === undefined)` ?

Reply:
that would ignore the `null` case